challenger/ExecutionPoker.js

```diff
@@ -0,0 +1,387 @@
+      this.log('goes right from ' +
+        obj.computationPath.hash.substring(2, 6) + ' to ' +
+        obj.computationPath.right.hash.substring(2, 6)
+      );
+    }
+
+    obj.computationPath = nextPath;
+
+    let witnessPath;
+
+    if (dispute.witness !== ZERO_HASH) {
+      const path = obj.merkle.getNode(dispute.witness);
+
+      witnessPath = { left: path.left.hash, right: path.right.hash };
+    } else {
+      witnessPath = { left: ZERO_HASH, right: ZERO_HASH };
+    }
+
+    // let d = await this.verifier.disputes(disputeId);
+    // console.log(disputeId, d, {
+    //   left: obj.computationPath.left.hash,
+    //   right: obj.computationPath.right.hash,
+    // }, witnessPath);
+    try {
+      let tx = await this.verifier.respond(
+        disputeId,
+        {
+          left: obj.computationPath.left.hash,
+          right: obj.computationPath.right.hash,
+        },
+        witnessPath,
+        { gasLimit: this.gasLimit, nonce: await this.getNonce() }
+      );
+
+      tx = await tx.wait();
+
+      this.log('gas used', tx.gasUsed.toString(), tx.transactionHash);
+    } catch (e) {
+      console.error('Submit round', e, {
+        left: obj.computationPath.left.hash,
+        right: obj.computationPath.right.hash,
+        witnessPath,
+        disputeId,
+      });
+    }
+  }
+
+  async submitProof (disputeId, disputeObj) {
+    const args = ProofHelper.constructProof(disputeObj.computationPath, disputeObj);
+
+    this.log('submitting proof - proofs', args.proofs);
+    this.log('submitting proof - executionState', args.executionInput);
+
+    try {
+      let tx = await this.verifier.submitProof(
+        disputeId,
+        args.proofs,
+        args.executionInput,
+        { gasLimit: this.gasLimit, nonce: await this.getNonce() }
+      );
+
+      tx = await tx.wait();
+
+      this.log('submitting proof - gas used', tx.gasUsed.toString(), tx.transactionHash);
+
+      return tx;
+    } catch (e) {
+      console.error('submitProff', e);
+    }
+  }
+
+  async computeCall (evmParams) {
+    let bytecode = await this.getCodeForParams(evmParams);
+    let data = await this.getDataForParams(evmParams);
+    let code = [];
+    let len = bytecode.length;
+
+    for (let i = 2; i < len;) {
+      code.push(bytecode.substring(i, i += 2));
+    }
+
+    let codeFragmentTree;
+    // code is not on chain-🍕
+    if (!evmParams.codeHash.endsWith('000000000000000000000000')) {
+      codeFragmentTree = new FragmentTree().run(bytecode);
+    }
+
+    const runtime = new HydratedRuntime();
+    const steps = await runtime.run({ code, data });
+    const merkle = new Merkelizer().run(steps, bytecode, data, evmParams.customEnvironmentHash);
+
+    return { steps, merkle, codeFragmentTree };
+  }
+
+  async getCodeForParams (evmParams) {
+    const addr = evmParams.codeHash.substring(0, 42);
+    return this.wallet.provider.getCode(addr);
+  }
+
+  async getDataForParams (evmParams) {
+    if (cliArgs.stupid) {
```

Comment:
Maybe this is the problem in https://github.com/leapdao/solEVM-enforcer/issues/163 ?

Reply:
No, it happens even without this flag